// mpackage/main.js
// module.exports.a=(name)=>{
//     console.log(name);
// }
export const a = (user)=>{
    const fi=(user.name.split(" "));
    const surname =firstLetter(fi[1]) 
    const name =firstLetter(fi[0])
    const purpose=firstLetter(user.purpose)
    console.log(purpose);
    const birth=user.dateBirth.split(".") 
    const d=(parseInt(birth[0]));
    const m=parseInt(birth[1])-1
    const y=parseInt(birth[2])
    const id = Math.random().toString(16).substring(2,12)
    console.log(id);
    const dataBirth=new Date(y,m,d)

    
    const yearUser=getYearUser(dataBirth)
    console.log(yearUser);
    return {
        id: id,  // случайны id из 10 символов 
        firstName: surname, // первая буква большая, остальные строчные
        lastName: name, // первая буква большая, остальные строчные
        dateBirth: dataBirth,
        age: yearUser, // высчитывается возраст на текущий день
        purpose: purpose, 
    }
}
const getYearUser=(dateUser)=>{
    const dateNow=new Date()

    const yearUser=(dateNow.getFullYear()-dateUser.getFullYear())
    const addY=(dateNow.getMonth()>dateUser.getMonth()) ?0:
    (dateNow.getMonth()<dateUser.getMonth()) ? 1:
    (dateNow.getDate()<dateUser.getDate()) ? 1:0
    return yearUser-addY
}
const firstLetter=(word)=>{
    return word[0].toUpperCase()+word.substring(1).toLowerCase()
}
// f1("gghg")
// module.exports=a
// module.exports ={a:5}
